### PR TITLE
reset default precision for datetime

### PR DESCRIPTION
**Pull request by ouyuanning**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

reset default precision to zero for DateTime type.  (was six before)

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by iamlinjunhong — APPROVED** [review]

File: test/result/expression/temporal_interval.result [hunk 1]
```diff
@@ -345,37 +345,37 @@ insert into t1 values(2,"1998-01-01","1998-01-01","1998-01-01 00:00:00.000001","
 insert into t1 values(3,NULL,NULL,NULL,NULL,NULL);
 select date_add(a,INTERVAL 1 SECOND), date_add(b,INTERVAL 1 MINUTE), date_add(c,INTERVAL 1 HOUR), date_add(d,INTERVAL 1 MONTH), date_add(e,INTERVAL 1 QUARTER) from t1;
 date_add(a,INTERVAL 1 SECOND)	date_add(b,INTERVAL 1 MINUTE)	date_add(c,INTERVAL 1 HOUR)	date_add(d,INTERVAL 1 MONTH)	date_add(e,INTERVAL 1 QUARTER)
-1997-12-31 00:00:01	1997-12-31 00:01:00	1998-01-01 00:59:59.000002000	1998-01-31 23:59:59	1998-03-31 23:59:59.000002000
-1998-01-01 00:00:01	1998-01-01 00:01:00	1998-01-01 01:00:00.000001000	1998-02-01 00:00:00	1998-03-31 23:59:59.000002000
+1997-12-31 00:00:01	1997-12-31 00:01:00	1998-01-01 00:59:59	1998-01-31 23:59:59	1998-03-31 23:59:59.000002000
```
Comment (fengttt) [inline test/result/expression/temporal_interval.result:348]: Did you check mysql behavior?   This feels strange -- precision is lost, and user has no idea that happened.

Please double check again with mysql, and product people.   We don't want to revert this later,

Comment (ouyuanning) [reply test/result/expression/temporal_interval.result:348]: Thank you for your advice. and i check with mysql。  
yes, there is some different:
1、MySQL's DateTime show the last unit is microsecond.  like : 
create table t1(i int,a date,b date,c datetime,d char(20),e varchar(50));
insert into t1 values(1,"1997-12-31","1997-12-31","1997-12-31 23:59:59.000002","1997-12-31 23:59:59","1997-12-31 23:59:59.000002");

select date_add(e,INTERVAL 1 QUARTER) from t1;
MySQL return : 1998-03-31 23:59:59.000002
MO    return : 1998-03-31 23:59:59.000002000

2、date_sub  datetime(with zero precision) with microsecond.  MySQL will get new precision with six。
  but MO still with zero。  like:
CREATE TABLE t1 (a DATETIME(6));
INSERT INTO t1 VALUES ('1000-01-01 01:00:00.000000');
SELECT DATE_SUB(CAST(a AS DATETIME), INTERVAL 1 MICROSECOND) FROM t1 ORDER BY a;
MySQL return: 1000-01-01 00:59:59
MO    return: 1000-01-01 00:59:59.999999

next week i will confirm with @domingozhang .

Comment (fengttt) [reply test/result/expression/temporal_interval.result:348]: We should not print out three 0s as nano seconds.   i dont think we have such precision.   

if there are microsecond (non zero) we should not omit.

Comment (ouyuanning) [reply test/result/expression/temporal_interval.result:348]: > ```sql
> CREATE TABLE t1 (a DATETIME(6));
> INSERT INTO t1 VALUES ('1000-01-01 01:00:00.000000');
> SELECT DATE_SUB(CAST(a AS DATETIME), INTERVAL 1 MICROSECOND) FROM t1 ORDER BY a;
> MySQL return: 1000-01-01 00:59:59.999999
> MO    return: 1000-01-01 00:59:59
> ```

My mistake.  this case
```sql
create table t1(i int,a date,b date,c datetime,d char(20),e varchar(50));
insert into t1 values(1,"1997-12-31","1997-12-31","1997-12-31 23:59:59.000002","1997-12-31 23:59:59","1997-12-31 23:59:59.000002");
select date_add(e,INTERVAL 1 QUARTER) from t1;
MySQL return : 1998-03-31 23:59:59.000002
MO    is also return : 1998-03-31 23:59:59.000002        
          -- if use mysql client in shell we get 1998-03-31 23:59:59.000002.  
          -- but in bvt test result we get 1998-03-31 23:59:59.000002000   ,  @daviszhen said that may be JDBC's error
```


and this case
```sql
CREATE TABLE t1 (a DATETIME(6));
INSERT INTO t1 VALUES ('1000-01-01 01:00:00.000000');
SELECT DATE_SUB(CAST(a AS DATETIME), INTERVAL 1 MICROSECOND) FROM t1 ORDER BY a;
MySQL return: 1000-01-01 00:59:59.999999
MO    return: 1000-01-01 00:59:59
```
@domingozhang : follow MySQL

**Review by bbbearxyz — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**